app/src/androidTest/java/com/mutualmobile/barricade/sample/utils/RecyclerViewMatcher.java

```diff
@@ -0,0 +1,67 @@
+package com.mutualmobile.barricade.sample.utils;
+
+import android.content.res.Resources;
+import android.support.v7.widget.RecyclerView;
+import android.view.View;
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.hamcrest.TypeSafeMatcher;
+
+/**
+ * Used for Matching contents of a RecyclerView item by position
+ */
+public class RecyclerViewMatcher {
+  private final int recyclerViewId;
+
+  public RecyclerViewMatcher(int recyclerViewId) {
+    this.recyclerViewId = recyclerViewId;
+  }
+
+  public Matcher<View> atPosition(final int position) {
+    return atPositionOnView(position, -1);
+  }
+
+  private Matcher<View> atPositionOnView(final int position, final int targetViewId) {
+
+    return new TypeSafeMatcher<View>() {
+      Resources resources = null;
+      View childView;
+
+      public void describeTo(Description description) {
+        String idDescription = Integer.toString(recyclerViewId);
+        if (this.resources != null) {
+          try {
+            idDescription = this.resources.getResourceName(recyclerViewId);
+          } catch (Resources.NotFoundException var4) {
+            idDescription = String.format("%s (resource name not found)", new Object[] {
```

Comment:
Can be simplified like this - `idDescription = String.format("%s (resource name not found)", recyclerViewId);`